src/persist-client/src/internal/state.rs

```diff
@@ -905,10 +920,17 @@ where
             applier_version: cfg.build_version.clone(),
             shard_id: self.shard_id,
             seqno: self.seqno.next(),
+            walltime_ms: (cfg.now)(),
             hostname: cfg.hostname.clone(),
             collections: self.collections.clone(),
             _phantom: PhantomData,
         };
+        // Make sure walltime_ms is strictly increasing, in case clocks are
```

Comment:
ah that's right! three motivations:

- speculative persistcli inspect debugging usefulness
- persist state is already a set of _definite_ collections (with a SeqNo timestmap), this makes it a _reclocked_ definite collection (context: #13918)
- if we want to do the read-only shard thing, we'll need some sort of "timestamp" hold where we give the read-only things a maximum of X amount of time that they can fall behind (read-only means we can't, and don't want to, let them participate in the current reader seqno hold stuff). this timestamp would be the basis of that

Reply:
For (1) it seems more useful to have the exact wallclock time of the handle, since we lose opportunities to observe clock skew in some cases otherwise. (2) is the need for it to be strictly increasing so that there's always a 1:1 mapping of real-time to SeqNo? (3) I'm on board with, I think we'll want a `walltime_ms` or equivalent for triggering orphaned blob maintenance at some point too

Reply:
For (2) we need a stronger property than 1:1, you can certainly pull the real math out of the reclocking RFC, but my summarized intuition is that the timestamp has to be strictly increasing as SeqNo increases. FWIW, my immediate motivation for the PR is (2) and (3), not (1), though the commit message and PR desc as they currently stand don't make that clear.